### ISSUE TITLE
Accessibility Issue when a checkbox has focus via a keyboard event such as Tab

### ISSUE THREAD

**Issue by mouse0270** [foundryvtt/foundryvtt#8655]
### User Experience

When an input element gains focus via tabbing or another method the user has not indication that the input is focus. Please add either an outline or box shadow on `:focus-visible`

To replicate this you should be able to just open up the module management window, click into the search field and then tab a few times to get focus on the first checkbox for the first module. It will look like this
![image](https://user-images.githubusercontent.com/564874/208256253-1a190a9d-e31b-4aec-9ee6-507d1259b1e8.png)

As you can see there is zero indication that this input has focus. This can be solved with some simple CSS for example you could do
```css
label.checkbox > input[type="checkbox"]:focus-visible {
    box-shadow: 0 0 5px var(--color-shadow-primary);
}
```
And you would get this as your result
![image](https://user-images.githubusercontent.com/564874/208256320-50517466-83a4-466d-abfe-31260cede38f.png)

**Comment by typhonrt** [comment]
I'm also using :focus-visible liberally for tab navigation / accessibility concerns in any custom components. I suggest `outline` is the correct / consistent visual cue in this case for explicit key navigation.

**Comment by mouse0270** [comment]
> I'm also using :focus-visible liberally for tab navigation / accessibility concerns in any custom components. I suggest `outline` is the correct / consistent visual cue in this case for explicit key navigation.

I can't find anything that specifically says `outline` is the *correct / consistent visual cue* queue. I've heard this said many times but honestly couldn't find any specific source. To me outline is much more limited in how you can style it and doesn't provide any direct accessibility that I can seem to figure out.

**Comment by typhonrt** [comment]
Lot's of articles suggesting outline...  Like this one:
https://www.sarasoueidan.com/blog/focus-indicators/#examining-the-accessibility-of-focus-indicators-in-popular-design-systems-and-ui-pattern-libraries

Search for `Avoid relying on background colors and box shadows alone to indicate focus.`

"Forced Color Modes and other accessibility display modes will override and/or remove them."

Using :focus-visible of course is the key IMHO as to why outline is more palatable of a solution as it requires explicit key navigation.

**Comment by mouse0270** [comment]
So outline is better for a specific accessibility and thats Forced Color modes. Even that article states this

> You can use backgrounds, borders and box shadows to style focus states. But if you do, add a transparent outline as a “fallback” for the various display modes:
```html
button:focus-visible {
	/* Default. Will be removed in Forced Color Modes */
	box-shadow: 5px 5px 7px rgba(0, 0, 0, 0.1); 
	/* Fallback. Will be visible with custom system colors in Forced Color Modes */
	outline: 1px solid transparent; 
}
```
Which makes sense and I'd simply change my stated CSS change to 
```css
label.checkbox > input[type="checkbox"]:focus-visible {
    box-shadow: 0 0 5px var(--color-shadow-primary);
    outline: 1px solid transparent; 
}
```
As IMHO, Outline is hideous design. I've always hated and is really basically dropped dropped it once `:focus-visible` became a thing, but to be honest, I also didn't think about `Forced Color Modes`. Which isn't something I would have ever thought of to be honest until someone pointed it out to me.

**Comment by typhonrt** [comment]
All of this of course IMHO... "Hideous design" is "accessible design". _Accessibility_ has different design constraints than more aesthetic concerns. Consistency across accessible / explicitly focused (keyboard) elements should be maintained and the browser / user agent default suffices which is `outline`. Also not all focusable elements will look correct w/ a `box-shadow` or make sense to have a `box-shadow` hence it's not great to have `box-shadow` as the primary `focus-visible` cue on some elements and not others.

Aspects like `box-shadow` should be used with `:focus` and `outline` should be used w/ `:focus-visible`.

Also consider the case of the checkbox. The user clicks on it making it the focused element and there is no cue that is such then spacebar is pressed and it is activated subsequently.  Consider a button or other input that is silently focused after the last mouse click that might do an undesirable action accidentally (deletion for instance). 

I very much like this idea in general as it accomplishes aesthetic concerns while still maintaining a consistent accessibility mode with explicit navigation:
https://css-tricks.com/the-focus-visible-trick/

In the case of the checkbox you can even take it a little further with CSS variables:

```css
   :root {
      --fvtt-input-box-shadow-focus: 0 0 5px var(--color-shadow-primary);
   }

   /* `revert` will use browser default if variable not defined */
   input[type="checkbox"]:focus-visible {
      box-shadow: var(--fvtt-box-shadow-focus-visible, none);
      outline: var(--fvtt-outline-focus-visible, revert);
   }

   input[type="checkbox"]:focus:not(:focus-visible) {
      box-shadow: var(--fvtt-input-box-shadow-focus);
      outline: none;
   }
```

This accomplishes the aesthetic goal and provides consistent accessibility cues with explicit keyboard. 

[checkbox-outline.webm](https://user-images.githubusercontent.com/311473/209225409-77a07722-2907-4395-8ef9-5edc2f1a0510.webm)

To offer a mode where `box-shadow` is applied for `:focus-visible` vs `outline` then setting the CSS variables to will swap things up:
```css
--fvtt-box-shadow-focus-visible: var(--fvtt-input-box-shadow-focus);
--fvtt-outline-focus-visible: none;
```

[checkbox-box-shadow.webm](https://user-images.githubusercontent.com/311473/209225082-2e34d55d-440b-417e-ba81-54a429ec2cab.webm)

-----------------------

It also should be noted that the Foundry default styles as currently defined throw a wrench into things with checkbox since it uses `--webkit-filter: greyscale(100%);`. This prevents `box-shadow` from being applied. Since the minimum browser requirements are moved up using `accent-color` is likely better as that allows `box-shadow` to be applied. 

I added the following CSS just to revert the foundry defaults: 
```css
   input[type="checkbox"] {
      accent-color: rgb(120, 120, 120);
      filter: none;
   }
```

-----------------------

In my UI framework / library I follow a similar pattern to the above with a default use of outline, but being able to change CSS variables to alter the default behavior.

**Comment by mouse0270** [comment]
+1 To actually using `accent-color` over `filter: greyscale(100%);`

Basically, to much detail about outline vs box shadow which are both accomplishing the same thing. I would suggest against using both methods at the same time and either choose to go with a mixed method in which for most users you use your preferred indicator to show that an input is focused, either outline or box shadow, however, if you choose to use a box shadow as your preferred method to indicate focus state, make sure to set `outline-color` to `transparent` and not `outline` to `none` as if you use `outline: none` you will be directly breaking forced color modes.

Just choose a method you feel is best for your design and go with it, because honestly both methods discussed here will meet accessibility standards and the rest of this conversation is about nit picking UI design. When really all I care about is adding a little more accessibility to the end users.

**Comment by typhonrt** [comment]
Just a note that `<input type="color">` also needs the same treatment.  Probably others, but haven't checked all input elements.
